fiat-sql/src/main/kotlin/com/netflix/spinnaker/fiat/permissions/SqlPermissionsRepository.kt

```diff
@@ -351,6 +312,23 @@ class SqlPermissionsRepository(
         }
     }
 
+    private fun addUserResources(userPermission: UserPermission) {
+        withRetry(RetryCategory.READ) {
+            jooq.selectFrom(RESOURCE.naturalJoin(PERMISSION))
```

Comment:
I'll note that the bulk of the optimization is in this line. When the permission and resource tables were initially split, foreign keys were added for the resource name and resource type to link these tables together. However, this query wasn't ever updated to make use of this natural join relation. Read-side chunking of the query is no longer necessary as we can let the database plan out how to chunk things itself when forming the row result set.